pkg/webhook/deployment.go

```diff
@@ -101,6 +103,34 @@ func getSpecDeployment(webhooksSecretName, namespace string) (*appsv1.Deployment
 							Image:           images.GetWebhookServerImage(),
 							Command:         []string{"/usr/local/bin/entrypoint", "/usr/local/bin/webhook-server"},
 							ImagePullPolicy: corev1.PullAlways,
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthz",
+										Port:   intstr.FromString("liveness-port"),
+										Scheme: "HTTP",
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+								PeriodSeconds:       1,
```

Comment:
The tricky part with probes, choose meaningful settings for timeouts. Controller and webhook server seems to have different values currently.
Webhook server has 1s delay + 1 second period * 3 times FailureThreshold. Does it mean that webhook server will be restarted in 4 seconds if probe don't get available for this pretty short time?

For liveness probe we have 30 times FailureThreshold with 1 second period, which means webhooks server will receive a traffic 30 seconds after it hangs or crashed (not the process but server itself is not able to handle traffic)
^ that it's not really true and just speculation since readiness probe and liveness probe use the same endpoint which means container will be restarted after 3 seconds of unavailability(1s period * 3 times FailureThreshold). But it indicates we are able to find better values.

I did not think carefully enough to propose good numbers.

Reply:
Nice find and good point! I've changed it to be more generous and made sure that both the numbers match for webhook server and the controller: https://github.com/devfile/devworkspace-operator/pull/309/files#diff-4585a303265f378412beec910445bdbc4cc40b71d51978eb4be67c4712faa2c6R18